bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/tracer/BSpan.java

```diff
@@ -15,151 +15,116 @@
+                .start();
+    }
+
+    private <C> BSpan(Format<C> format, C carrier, String serviceName, String operationName, boolean isClientContext) {
+        tracer = TracersStore.getInstance().getTracer(serviceName);
+        SpanContext parentSpanContext = tracer.extract(format, carrier);
+        span = tracer.buildSpan(operationName)
+                .asChildOf(parentSpanContext)
+                .withTag(TraceConstants.TAG_KEY_SPAN_KIND, isClientContext
+                        ? TraceConstants.TAG_SPAN_KIND_CLIENT
+                        : TraceConstants.TAG_SPAN_KIND_SERVER)
+                .start();
+    }
 
     /**
-     * {@link Map} of properties, which used to represent
-     * the span contexts of each tracer.
+     * Start a new span without a parent. The started span will be a root span.
+     *
+     * @param serviceName   The name of the service the span belongs to
+     * @param operationName The name of the operation the span corresponds to
+     * @param isClient      True if this is a client span
+     * @return The new span
      */
-    private final Map<String, String> properties;
-    /**
-     * {@link Map} of tags, which will get injected to span.
-     */
-    private final Map<String, String> tags;
-    /**
-     * Name of the service.
-     */
-    private String serviceName = DEFAULT_SERVICE_NAME;
-    /**
-     * Name of the operation.
-     */
-    private String operationName = DEFAULT_OPERATION_NAME;
-    /**
-     * Active Ballerina {@link ObserverContext}.
-     */
-    private final ObserverContext observerContext;
+    public static BSpan start(String serviceName, String operationName, boolean isClient) {
+        return new BSpan(null, serviceName, operationName, isClient);
+    }
+
     /**
-     * Open tracer specific span.
+     * Start a new with a parent using parent span.
+     *
+     * @param parentSpan    The parent span of the new span
+     * @param serviceName   The name of the service the span belongs to
+     * @param operationName The name of the operation the span corresponds to
+     * @param isClient      True if this is a client span
+     * @return The new span
      */
-    private Span span;
-
-    public BSpan(ObserverContext observerContext, boolean isClientContext) {
-        this.properties = new HashMap<>();
-        this.tags = new HashMap<>();
-        this.observerContext = observerContext;
-        this.tags.put(TraceConstants.TAG_KEY_SPAN_KIND, isClientContext
-                ? TraceConstants.TAG_SPAN_KIND_CLIENT
-                : TraceConstants.TAG_SPAN_KIND_SERVER);
+    public static BSpan start(BSpan parentSpan, String serviceName, String operationName, boolean isClient) {
+        return new BSpan(parentSpan.span.context(), serviceName, operationName, isClient);
     }
 
-    public void startSpan() {
-        manager.startSpan(getParentBSpan(), this);
+    /**
+     * Start a new with a parent using parent trace context.
```

Comment:
```suggestion
     * Start a new span with a parent using parent trace context.
```

Reply:
Fixed in fc0e0f80fdfa26ccbb5cdeb99deac877c0bb8261